### ISSUE TITLE
caffe('get_init_key')   Undefined function 'caffe' for input arguments of type 'char'.

### ISSUE THREAD

**Issue by HuntingGo** [rbgirshick/rcnn#70]
i installed caffe, and clone the rcnn,  and i also run the startup.m  rcnn_build.m with no error ,but when i try to run key = caffe('get_init_key'), i got the following message.

>> caffe('get_init_key') 
Undefined function 'caffe' for input arguments of type 'char'.
 
I can use matcaffe like caffe.set_mode_gpu(), but I can't use caffe('set_mode_gpu').
anybody can give me the appropriate solution for the problem
Please help.

**Comment by jainanshul** [comment]
Were you able to get answers to your question? I am hitting the same issue and not sure what the solution is.

**Comment by HuntingGo** [comment]
@jainanshul  yeah. Using the caffe of current version may be a leading cause of errors. U can try the following version of caffe: https://github.com/BVLC/caffe/archive/v0.999.tar.gz

**Comment by jainanshul** [comment]
Thanks, I got it working withe v0.999 version.

**Comment by ducminhnguyen** [comment]
I cannot compile make matcaffe for 0.999 it kept printing linker error from libcaffe.a

**Comment by WideLee** [comment]
I solve it by link the `caffe_.mexa64` which is build by `make matcaffe` to the folder `rcnn/bin` by below command:
``` bash
$ ln -s $CAFFE_ROOT/matlab/caffe/private/caffe_.mexa64 $RCNN_ROOT/bin/caffe.mexa64
```

After make a symlink like this, caffe is found by matlab, but my `caffe('get_init_key')` get an random key id such as 2.6446e+09, and it is not equal to -2 which is referred in readme file, I don't know why T^T . Now I'm downing pre-trained model and try to run demo of rcnn.

Hope I can help you ^_^

**Comment by dorioliu** [comment]
@WideLee   HI ,did you  successfully  run demo of rcnn? I meet  the same question. thanks!

**Comment by WideLee** [comment]
@dorioliu Have you tried the method I mentioned above?

**Comment by dorioliu** [comment]
@WideLee  sorry ,i havn't tried it ,did you successfully run demo of rcnn through you got a strange caffe('get_init_key') value ? i will try it latter.

**Comment by WideLee** [comment]
@dorioliu It's a year ago, I'm not sure right now. You can try [Faster R-CNN](https://github.com/rbgirshick/py-faster-rcnn) or [Mask R-CNN](https://github.com/matterport/Mask_RCNN) for faster and more accurate result. 😄
